### ISSUE TITLE
Exposing native device ID 

### ISSUE THREAD

**Issue by fnrizzi** [kokkos/kokkos#6040]
Potential improvement to #5855 would be to expose the native device ID. 
That PR is not including this change as per discussion with @PhilMiller @dalg24 , so this is left for future addition.

**Comment by crtrott** [comment]
I think that is a backend specific thing, and should be backend specific queries such as "Cuda().get_cuda_config()" or "SYCL().get_sycl_config()". If we do anything like that I think this needs to be extremely clear that it is interoperability stuff and you are specific to a specific backend. Defacto we already got "cuda_stream()" "hip_stream()" and "sycl_queue()" not sure we need more at this point?

**Comment by fnrizzi** [comment]
@crtrott yes makes sense about being very clear. I opened as placeholder following discussion with Damien and Phil, happy to brainstorm more on this.

**Comment by ajpowelsnl** [comment]
@fnrizzi -- should this issue be a topic for Wednesday discussion?

**Comment by ajpowelsnl** [comment]
@fnrizzi - adding this issue to the Kokkos Planning Board
